ui/src/CollectionExerciseDetails.js

```diff
@@ -87,6 +124,37 @@ class CollectionExerciseDetails extends Component {
       failedValidation = true
     }
 
+    if (!this.state.newWaveOfContactClassifiers.trim()) {
+      this.setState({ classifiersValidationError: true })
+      failedValidation = true
+    }
+
+    if (!this.state.newWaveOfContactTemplate.trim()) {
+      this.setState({ templateValidationError: true })
+      failedValidation = true
+    } else {
+      try {
+        const parsedJson = JSON.parse(this.state.newWaveOfContactTemplate)
```

Comment:
Forgot to say - could probably refactor these bits into helper functions or something. One for the future I guess? Will raise a small tech ticket